File: sample/src/main/java/org/duncavage/swipetorefresh/sample/MainActivity.java
package org.duncavage.swipetorefresh.sample;

import android.os.Handler;
import android.support.v7.app.ActionBarActivity;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;

import org.duncavage.swipetorefresh.widget.SwipeRefreshLayout;

public class MainActivity extends ActionBarActivity implements SwipeRefreshLayout.OnRefreshListener{
    private SwipeRefreshLayout mSwipeRefreshLayout;
    private Handler mHandler;

    private final Runnable mStopRefresh = new Runnable() {
        @Override
        public void run() {
            mSwipeRefreshLayout.setRefreshing(false);
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        mHandler = new Handler();

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mSwipeRefreshLayout = (SwipeRefreshLayout)findViewById(R.id.swipe_refresh_layout);
        mSwipeRefreshLayout.setOnRefreshListener(this);
        // To use the default layout:
        // Set the background color
        mSwipeRefreshLayout.setActionBarSwipeIndicatorBackgroundColor(
                getResources().getColor(R.color.swipe_to_refresh_background));
        // Set the text colors. If you don't the text will not display.
        mSwipeRefreshLayout.setActionBarSwipeIndicatorTextColor(
                getResources().getColor(R.color.swipe_to_refresh_text));
        mSwipeRefreshLayout.setActionBarSwipeIndicatorRefreshingTextColor(
                getResources().getColor(R.color.swipe_to_refresh_text));

        // Or you can use a custom layout...
        // This is recommended if you want more control over the text styling.
        // mSwipeRefreshLayout.setActionBarSwipeIndicatorLayout(R.layout.swipe_indicator, R.id.text);

        mSwipeRefreshLayout.setActionBarSwipeIndicatorText(R.string.swipe_to_refresh);
        mSwipeRefreshLayout.setActionBarSwipeIndicatorRefreshingText(R.string.loading);
        mSwipeRefreshLayout.setColorScheme(
                R.color.refreshing_color1,
                R.color.refreshing_color2,
                R.color.refreshing_color3,
                R.color.refreshing_color4
        );
    }

    @Override
    public void onRefresh() {
        mHandler.postDelayed(mStopRefresh, 5000);
    }
}

File: csharp/SwipeToRefresharpSample/SwipeToRefresharpSample/MainActivity.cs
using System;
using Android.App;
using Android.Content;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Android.OS;
using SwipeToRefresh.Widget;

namespace SwipeToRefresharpSample
{
    [Activity(Label = "SwipeToRefresharpSample", MainLauncher = true, Theme = "@style/AppTheme")]
    public class MainActivity : Activity
    {
        Handler handler;
        SwipeRefreshLayout swipe_refresh_layout;

        protected override void OnCreate(Bundle bundle)
        {
            handler = new Handler();

            base.OnCreate(bundle);

            // Set our view from the "main" layout resource
            SetContentView(Resource.Layout.Main);

            swipe_refresh_layout = FindViewById<SwipeRefreshLayout>(Resource.Id.swipe_refresh_layout);
            // Setup the actionbar indicator

            // To use the default layout:
            // Set the background color
            swipe_refresh_layout.SetActionBarSwipeIndicatorBackgroundColor(Resources.GetColor(Resource.Color.swipe_to_refresh_background));
            // Set the text colors. Failing to do so will cause the text to not be displayed.
            swipe_refresh_layout.SetActionBarSwipeIndicatorTextColor(Resources.GetColor(Resource.Color.swipe_to_refresh_text));
            swipe_refresh_layout.SetActionBarSwipeIndicatorRefreshingTextColor(Resources.GetColor(Resource.Color.swipe_to_refresh_text));

            // Or you can use a custom layout...
            // This is recommended if you want more control over the text styling.
            //swipe_refresh_layout.SetActionBarSwipeIndicatorLayout(Resource.Layout.swipe_indicator, Resource.Id.text);

            // Set the text to be displayed.
            swipe_refresh_layout.SetActionBarSwipeIndicatorRefreshingText(Resource.String.loading);
            swipe_refresh_layout.SetActionBarSwipeIndicatorText(Resource.String.swipe_to_refresh);

            // Setup colors
            swipe_refresh_layout.SetColorScheme(
                Resource.Color.refreshing_color1,
                Resource.Color.refreshing_color2,
                Resource.Color.refreshing_color3,
                Resource.Color.refreshing_color4);
            // Set the listener
            swipe_refresh_layout.Refresh += OnRefresh;
        }

        private void OnRefresh(object sender, EventArgs args)
        {
            handler.PostDelayed(() => {
                swipe_refresh_layout.Refreshing = false;
            }, 3000);
        }
    }
}


